src/ray/raylet/scheduling/cluster_task_manager.h

```diff
@@ -46,14 +47,13 @@ class ClusterTaskManager {
   /// \param self_node_id: ID of local node.
   /// \param cluster_resource_scheduler: The resource scheduler which contains
   /// the state of the cluster.
-  /// \param fulfills_dependencies_func: Returns true if all of a task's
-  /// dependencies are fulfilled.
+  /// \param task_dependency_manager_ Used to fetch task's dependencies.
   /// \param is_owner_alive: A callback which returns if the owner process is alive
   /// (according to our ownership model).
   /// \param gcs_client: A gcs client.
   ClusterTaskManager(const NodeID &self_node_id,
                      std::shared_ptr<ClusterResourceScheduler> cluster_resource_scheduler,
-                     std::function<bool(const Task &)> fulfills_dependencies_func,
+                     TaskDependencyManagerInterface &task_dependency_manager_,
```

Comment:
Adding this interface / persistent subscriptions until dispatch sounds good :+1: 

To avoid needing adding TasksBlocked (edge-triggered logic), could we instead double check the dependencies prior to dispatch of a task (level-triggered)?